pkg/auth/webapp/session_authflow.go

```diff
@@ -840,15 +841,23 @@ func (s *AuthflowScreenWithFlowResponse) makeVerifyOOBOTPOutputTransformer(chann
 			startTime := deps.Clock.NowUTC()
 
 			for {
+				data := output.FlowAction.Data.(declarative.VerifyOOBOTPData)
 				now := deps.Clock.NowUTC()
 				// Normally, the timeout should be handled by the whatsapp callback timeout.
 				// Therefore we should always get failed / sent status within the configured timeout.
 				// So this timeout is just as a last resort to break the loop to avoid waiting forever.
 				if now.Sub(startTime) > messageDeliveryStatusMaxWaitTime {
-					return nil, fmt.Errorf("timed out waiting for delivery_status")
+					return nil, errorutil.WithDetails(fmt.Errorf("timed out waiting for delivery_status"),
+						errorutil.Details{
+							"otp_form":                           data.OTPForm,
+							"channel":                            data.Channel,
+							"failed_attempt_rate_limit_exceeded": data.FailedAttemptRateLimitExceeded,
+							"delivery_status":                    data.DeliveryStatus,
+							"delivery_error":                     data.DeliveryError,
+						},
+					)
```

Comment:
It seems that details cannot be logged. https://oursky.slack.com/archives/CUUDP0XJQ/p1761230768391579?thread_ts=1761230129.453129&cid=CUUDP0XJQ

Reply:
Created https://linear.app/authgear/issue/DEV-3122/handle-tagged-detail-in-logger for the issue you mentioned